server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedDecoder.java

```diff
@@ -1105,6 +1132,14 @@ private MemcachedOperation toRequest(String commandName, Boolean endOfOp, ByteBu
             throw new UnknownOperationException("Unknown operation: " + commandName);
       }
    }
+
+   private OperationContext createOperationContext() {
+      return new OperationContext(header, params, key, rawValue);
+   }
+
+   private void sendResponseOrdered(Channel ch, CompletionStage<Object> rsp) {
+      new ResponseEntry(this, ch).queueResponse(rsp);
```

Comment:
This makes me think we should add a test that does 2 operations in parallel with the non blocking mecached API. Or do we already have that?

Reply:
I can remove the `get()` from here: https://github.com/infinispan/infinispan/pull/10379/files#diff-49239ed285f72772402428f971aa797f4a8be001e4cf22c5a31765afe0e207ebR50

The test only needs the last `get()` to make sure the `flush()` is not executed concurrently with the puts.

Reply:
I did the change above ^

Reply:
Only problem is that we aren't checking the response and they aren't interesting in the test. I would say we should do this to a test that does two gets in parallel that get different values in the response, just to make sure the responses aren't interleaved.

Reply:
added `org.infinispan.server.functional.MemcachedOperations#testConcurrentGets`

Reply:
Great, thanks 👍